### PR TITLE
txnbuild: add SequenceNumber func to Transaction

### PR DESCRIPTION
**Pull request by leighmcculloch**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What
Add `SequenceNumber` function to `Transaction`.

### Why
Stellar transactions have a sequence number. If a transaction is valid and it is
accepted into a ledger the source account of the transaction is updated such
that its sequence number becomes the sequence number of the transaction.

In the Go SDK a parsed transaction has no sequence number field or function
directly on it. For a developer to get the sequence number they must look at the
source account object. This is not intuitive and not consistent conceptually
because a transaction has a sequence number and it should be a first class
attribute of the transaction.

In my own development using the Go SDK I spent time trying to figure out where
the sequence was stored and how to get at it. I work on Stellar everyday, so if
this is not intuitive to me, I suspect it isn't intuitive to others.

Other Stellar SDKs, such as the JS SDK, expose it as an attribute on a
Transaction.

This is likely to become even less intuitive as we plan to change the sequence
number rules where the sequence number on an account does not need to be
sequential. Regardless of this, it isn't intuitive today.

### Known limitations

N/A

### CODE REVIEW

**Review by acharb — APPROVED** [review]
lgtm!

File: txnbuild/transaction.go [hunk 1]
```diff
@@ -212,6 +212,11 @@ func (t *Transaction) SourceAccount() SimpleAccount {
 	return t.sourceAccount
 }
 
+// SequenceNumber returns the sequence number of the transaction.
+func (t *Transaction) SequenceNumber() int64 {
+	return t.sourceAccount.Sequence
+}
+
```
Comment (marcelosalloum) [inline txnbuild/transaction.go:219]: ❓ Shouldn't we return `t.sourceAccount.Sequence + 1` when `t.IncrementSequenceNumber` is true?

**Review by marcelosalloum — COMMENTED** [review]
Shouldn't the transaction sequence number depend on the `IncrementSequenceNumber` flag as well?

File: txnbuild/transaction.go (hunk 1 above)
Comment (leighmcculloch) [reply txnbuild/transaction.go:219]: The `Transaction` type has no `IncrementSequenceNumber` field or function. I think you're thinking of `TransactionParams` which has the confusing and unexpected-side-effect-causing `IncrementSequenceNumber`.

Comment (marcelosalloum) [reply txnbuild/transaction.go:219]: Cool, thanks for explaining.

**Review by ire-and-curses — APPROVED** [review]
Great, this looks really helpful, especially for any non-trivial application (i.e. one that assumes validity, and tracks sequence numbers locally rather than looking up Horizon each time).

**Pull request merged**